### DISCUSSION TITLE
Preview Entries in the CP that Aren't Available Publicly

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#11376] [Q&A] [answered]
Does anyone have a solution for how I can preview an entry within the control panel for content that I never want to be viewable directly from a public url?

I have a section of entries that are effectively individual building blocks for a page.  I would like the ability to preview the individual entries in the control panel but only allow them to be viewed outside of the control panel if they are represented as a "block" within a page.

**Comment by wsydney76** [comment, marked as answer]
Found some old stuff that should work for that (at least as a starting point...)

Set up a preview target with a URL format like `/internal/sections/pageblock?id={id}`

The template can now check whether it is in preview mode and retrieve the entry:

```
{% requireLogin %}

{% if not craft.app.request.isPreview %}
    {% exit 404 %}
{% endif %}

{% set id = craft.app.request.param('id') %}

{% if not id %}
    {% exit 404 %}
{% endif %}

{# That should fetch an entry regardless of status/provisionalDraft/draft whatever ?? #}
{% set entry = craft.app.entries.getEntryById(id) %}

{% if not entry %}
    {% exit 404 %}
{% endif %}

<h1>{{ entry.title }}</h1>

<h2>ID: {{ entry.id }}</h2>
```

/Mel

**Reply by brandonkelly** [reply]
Yep this is exactly what Preview Targets are for :)
